backend/app/tasks/subscription_tasks.py

```
@@ -464,6 +472,33 @@ async def _create_subscription_task(
+                if execution_target.type == SubscriptionExecutionTargetType.LOCAL
+                else DeviceType.CLOUD
+            )
+            default_device = device_service.get_default_device_for_type(
+                db,
+                user_id=ctx.user.id,
+                device_type=target_device_type,
+            )
+            if not default_device:
+                raise ValueError(
+                    f"No default {execution_target.type.value} device available for subscription {ctx.subscription.id}"
+                )
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)